src/collection-preferences/visible-content.tsx

```diff
@@ -78,21 +101,34 @@ export default function VisibleContentPreference({
         {options.map((optionGroup, optionGroupIndex) => {
           const groupLabelId = `${idPrefix}-${optionGroupIndex}`;
           return (
-            <div
+            <DndContext
               key={optionGroupIndex}
-              {...className('group')}
-              role="group"
-              aria-labelledby={`${outerGroupLabelId} ${groupLabelId}`}
+              sensors={sensors}
+              collisionDetection={closestCenter}
+              onDragEnd={event => {
+                const { active, over } = event;
+
+                if (over && active.id !== over.id) {
+                  setSortedOptions(items => {
+                    const oldIndex = items.findIndex(option => option.id === active.id);
+                    const newIndex = items.findIndex(option => option.id === over.id);
+
+                    return arrayMove(items, oldIndex, newIndex);
+                  });
+                }
+              }}
             >
               <div {...className('group-label')} id={groupLabelId}>
                 {optionGroup.label}
               </div>
-              <div>
-                {optionGroup.options.map((option, optionIndex) =>
-                  selectionOption(option, optionGroupIndex, optionIndex)
-                )}
-              </div>
-            </div>
+              <SortableContext items={sortedOptions} strategy={verticalListSortingStrategy}>
```

Comment:
Using `rectSortingStrategy` instead of `verticalSortingStrategy` here fixes the overlaps that occur because the items that move as a "side effect" (i.e, not the directly dragged item) don't change column until dropping.

https://user-images.githubusercontent.com/1257272/223122782-f457ab41-4e41-4e0f-9acc-8e09ad86f3e8.mov